internal/options/alias.go

```diff
@@ -0,0 +1,246 @@
+package options
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/iancoleman/strcase"
+	"github.com/launchdarkly/ld-find-code-refs/internal/validation"
+	"gopkg.in/yaml.v2"
+)
+
+// We're using multiple packages for configuration.
+// TODO: Use spf13/viper to standardize configuration for args, env, and yaml.
+
+type AliasType string
+
+func (a AliasType) IsValid() error {
+	switch a {
+	case Literal, CamelCase, PascalCase, SnakeCase, UpperSnakeCase, KebabCase, DotCase, FilePattern, Command:
+		return nil
+	}
+	return fmt.Errorf("'%s' is not a valid alias type", a)
+}
+
+func (t AliasType) unexpectedFieldErr(field string) error {
+	return fmt.Errorf("unexpected field for %s alias: '%s'", t, field)
+}
+
+func (a Alias) Generate(flag string) ([]string, error) {
+	ret := []string{}
+	switch a.Type {
+	case Literal:
+		ret = a.Flags[flag]
+	case CamelCase:
+		ret = []string{strcase.ToLowerCamel(flag)}
+	case PascalCase:
+		ret = []string{strcase.ToCamel(flag)}
+	case SnakeCase:
+		ret = []string{strcase.ToSnake(flag)}
+	case UpperSnakeCase:
+		ret = []string{strcase.ToScreamingSnake(flag)}
+	case KebabCase:
+		ret = []string{strcase.ToKebab(flag)}
+	case DotCase:
+		ret = []string{strcase.ToDelimited(flag, '.')}
+	case FilePattern:
+		pattern := regexp.MustCompile(strings.ReplaceAll(*a.Pattern, "FLAG_KEY", flag))
```

Comment:
Using `MustCompile` is safe because we already validated the regular expression at initialization time